mc2skos/record.py

```diff
@@ -266,6 +274,63 @@ def parse(self, options):
     def is_public(self):
         return True
 
+    def append_relation(self, scheme_code, scheme_type, relation, **kwargs):
+        try:
+            scheme = ConceptScheme(scheme_type, scheme_code, edition=kwargs.get('edition'))
+            uri = scheme.get_uri(**kwargs)
+        except UnknownSchemeError as e:
+            logger.warning('Cannot generate URIs for unknown vocabulary "%s"', scheme_code)
+            return
+
+        if uri:
+            self.relations.append({
+                'uri': uri,
+                'relation': relation,
+            })
+
+    def get_mappings(self):
+        for heading in self.get_terms('7'):
+            relation = None
+            for sf in heading['node'].all('mx:subfield'):
+                if sf.get('code') == '4':
+                    if is_uri(sf.text()):
+                        relation = URIRef(sf.text())
+                    else:
+                        relation = {
+                            '=EQ': SKOS.exactMatch,
+                            '~EQ': SKOS.closeMatch,
+                            'BM': SKOS.broadMatch,
+                            'NM': SKOS.narrowMatch,
+                            'RM': SKOS.relatedMatch,
+                        }.get(sf.text())  # None if no match
+
+                elif sf.get('code') == '0':
+                    # Note: Default value might change in the future
+                    relation = relation if relation else SKOS.closeMatch
```

Comment:
I realize this is just my own code moved to a new place, but got me thinking about the order of subfields. This code currently requires $4 to come before $0, not after. Is that a valid assumption to make, or could it equally well be the other way around?

Reply:
I guess it could also be the other way - if subfields are not repeatable, the order should be irrelevant.

Reply:
The problem is that they *are* repeatable. But I have also yet to see it in practice, and I'm not sure what it would *mean*.